### ISSUE TITLE
Erroneous "Relational comparison with number may involve unintended type conversions" with optional property

### ISSUE THREAD

**Issue by cyounkins** [facebook/flow#281]
Example: http://tryflow.org/#7278931169c2164c2513b61c3f111136

```
/* @flow */
 
type getOptions = {retries?: number};
var get = function(opts: getOptions): void {
        if (opts.retries === undefined) {
            opts.retries = 0;
        }
 
        if (opts.retries > 3) { // <--
 
        }
};
```

Since there is an undefined check, opts.retries must be a number. Thus, the relational comparison is between two numbers.

**Comment by jeffmo** [comment]
The existence of the error addressed here: https://github.com/facebook/flow/issues/282

That error message is pretty horrible though, and we should definitely make that less cryptic.

**Comment by jeffmo** [comment]
Here's where the error message is defined for whoever picks this up:
https://github.com/facebook/flow/blob/master/src/typing/flow_js.ml#L2446

**Comment by cyounkins** [comment]
I don't believe this is related to #282 - The error messages are different, and #282 specifically involves going into a function.

**Comment by jeffmo** [comment]
Ah yes, you're right -- though I think the two might be related in terms of how feasible it is to refine in this way.

I'll follow up with @avikchaudhuri  and/or @bhosmer to see what they think (they should have some more insight into how easy/hard this would be to accomplish).

**Comment by mroch** [comment]
since this was filed, this bug was fixed and then broke again, and then was fixed again most recently by a45a447ad8357bd952b8f61a9885cc990cdd059d. should be fixed in 0.19. :)
